File: CCC/03s3.java
import java.util.*;
public class FloorPlan {

	static int R;
	static int C;
	static char[][] grid;
	static int count;
	static ArrayList<Integer> size;
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int area = sc.nextInt();
		R = sc.nextInt();
		C = sc.nextInt();
		count = 0;
		int rooms = 0;
		grid = new char[R][C];
		size = new ArrayList<>();
		sc.nextLine();
		
		for(int i=0;i<R;i++) {
			String s = sc.nextLine();
			for(int j=0;j<C;j++) {
				grid[i][j] = s.charAt(j);
			}
		}
		
		for(int i=0;i<R;i++) {
			for(int j=0;j<C;j++) {
				if(grid[i][j] == '.') {
					size.add(fill(i, j));
				}
			}
		}
		Collections.sort(size, Collections.reverseOrder());	
		//System.out.println(size.toString());
		
		int f = 0;
		for(int i=0;i<size.size();i++) {
			int get = size.get(i);
			//System.out.println(get);
			if(f+get <= area) {
				f += get;
				rooms++;
				//System.out.println(f+" "+rooms+" "+get);
			}
			else {
				break;
			}
		}
		
		if(rooms == 1) {
			System.out.println(rooms+" room, "+(area-f)+" square metre(s) left over");
		}
		else {
			System.out.println(rooms+" rooms, "+(area-f)+" square metre(s) left over");
		}

	}
	public static int fill(int row, int col) {
		if(row < 0 || col < 0 || row >= R || col >= C) {
			return 0;
		}
		if(grid[row][col] == 'I') {
			return 0;
		}
		grid[row][col] = 'I';
		count = 1;
		count = count+ fill(row+1, col);	//check down
		count = count+ fill(row-1, col);	//check up
		count = count+ fill(row, col+1);	//check right side
		count = count+ fill(row, col-1);	//check left side
		return count;
	}


}

File: CCC/03s3.py
def floodfill(row, col):
    if row<0 or col<0 or row >=R or col>= C:
        return 0
    if floor[row][col] == "I":
        return 0
    floor[row][col] = "I"
    count = 1
    count = count+floodfill(row+1, col)
    count = count+floodfill(row-1, col)
    count = count+floodfill(row, col+1)
    count = count+floodfill(row, col-1)
    return count
 
sqm = int(input())
R = int(input())
C = int(input())
 
floor = []
for m in range(R):
    onerow = [x for x in input()]
    floor.append(onerow)
 
room = []
for x in range(R):
    for y in range(C):
        if floor[x][y] == '.':
            room.append(floodfill(x, y))
 
 
room.sort(reverse=True)

count = 0
numOfRooms = 0
for x in range(len(room)):
    if count+room[x] <= sqm:  
        count = count+room[x]
        numOfRooms += 1
    else:
        break
 
if numOfRooms == 1:
    print(str(numOfRooms)+" room, "+ str(sqm-count) +" square metre(s) left over")
else:
    print(str(numOfRooms)+" rooms, "+ str(sqm-count) +" square metre(s) left over")
